src/resources/platform_handlers/pytorch_libtorch/model.py

```diff
@@ -0,0 +1,241 @@
+
+        kind = args["model_instance_kind"]
+        device_id = args["model_instance_device_id"]
+        self._device = torch.device(_get_device_name(kind, device_id))
+
+        model_path = _get_model_path(self._model_config)
+        if not _is_py_class_model(model_path):
+            self._logger.log_info("Loading '" + self._model_name + "' as TorchScript")
+            self._model = torch.jit.load(model_path)
+            self._model.to(self._device)
+            self._model.eval()
+            return
+
+        self._model_module = _import_module_from_path(self._model_name, model_path)
+        self._model_class = _get_model_class_from_module(self._model_module)
+        self._raw_model = self._model_class()
+        data_path = _get_model_data_path(model_path)
+        if data_path != "":
+            self._raw_model.load_state_dict(
+                torch.load(data_path, map_location=self._device)
+            )
+        else:
+            self._logger.log_info(
+                "Model parameter file not found for '" + self._model_name + "'"
+            )
+        self._raw_model.to(self._device)
+        self._raw_model.eval()
+        if _enable_torch_compile(self._model_config):
+            self._model = torch.compile(self._raw_model)
+        else:
+            self._logger.log_info(
+                "'torch.compile' is disabled for '" + self._model_name + "'"
+            )
+            self._model = self._raw_model
+
+    def execute(self, requests):
+        """`execute` MUST be implemented in every Python model. `execute`
+        function receives a list of pb_utils.InferenceRequest as the only
+        argument. This function is called when an inference request is made
+        for this model. Depending on the batching configuration (e.g. Dynamic
+        Batching) used, `requests` may contain multiple requests. Every
+        Python model, must create one pb_utils.InferenceResponse for every
+        pb_utils.InferenceRequest in `requests`. If there is an error, you can
+        set the error argument when creating a pb_utils.InferenceResponse
+
+        Parameters
+        ----------
+        requests : list
+          A list of pb_utils.InferenceRequest
+
+        Returns
+        -------
+        list
+          A list of pb_utils.InferenceResponse. The length of this list must
+          be the same as `requests`
+        """
+        responses = []
+
+        # FIXME: Instead of iterating through each request, run the inference as a single batch.
+        for request in requests:
+            input_tensors = []
+            for io in self._inputs:
+                tensor = pb_utils.get_input_tensor_by_name(
+                    request, io["name"]
+                ).to_dlpack()
+                tensor = torch.from_dlpack(tensor).to(self._device)
+                input_tensors.append(tensor)
+
+            raw_output_tensors = self._model(*input_tensors)
```

Comment:
We should use inference_mode optimization for inference runs. Read more here: https://pytorch.org/docs/stable/generated/torch.inference_mode.html

Reply:
Yes. Added.